### DISCUSSION TITLE
Share intializationUrls

### DISCUSSION THREAD

**Discussion by glughi** [TerriaJS/terriajs#6398] [Q&A]
Hello,
there is a way to share (and load with `#start`) the JSON catalog used?
Thanks in advance,
G.

**Comment by nf-s** [comment]
Hi @glughi 

Sorry I'm not sure I totally understand your question. 

Are trying to load JSON catalogs through URL fragments (hash parameters)? If so there are two methods:

- `#initUrl` 
- `#start=shareData`

But if you are trying to create share links with maps that use either of these - the URL fragments won't be shared.

## `#initUrl`

Where `initUrl` is URL to a JSON Catalog

For example

- http://ci.terria.io/main/#https://gist.githubusercontent.com/nf-s/67e44309245f3caca1cee5cdf2eee3eb/raw/75c50871fd972f8d6f35039c6e5278f3f7437a69/colorscale-minmax.json

Is loading the map

- https://ci.terria.io/main/

With `initUrl`

- http://gist.githubusercontent.com/nf-s/67e44309245f3caca1cee5cdf2eee3eb/raw/75c50871fd972f8d6f35039c6e5278f3f7437a69/colorscale-minmax.json

## `#start=shareData`

[Share data JSON](https://github.com/TerriaJS/terriajs/blob/main/doc/customizing/initialization-files.md#sharedata) is different from catalog JSON

It has two properties:
- `version` which must be `8.0.0`
- `initSources` - which is equivalent to an array of JSON catalog files (but also supports URLs and "init fragments")

For example

```json
{
  "version": "8.0.0",
  "initSources": [
    {
      "catalog": [
        {
          "type": "csv",
          "url": "test/incidents_notime.csv",
          "name": "start test"
        }
      ]
    }
  ]
}
```

As the JSON blob is going into a URL, it must be encoded like so:

-  `{%22version%22:%20%228.0.0%22,%22initSources%22:%20[{%22catalog%22:%20[{%22type%22:%20%22csv%22,%22url%22:%20%22test/incidents_notime.csv%22,%22name%22:%20%22start%20test%22}]}]}`

Then attached to a Terria Map like `http://ci.terria.io/main/#start=`

- http://ci.terria.io/main/#start={%22version%22:%20%228.0.0%22,%22initSources%22:%20[{%22catalog%22:%20[{%22type%22:%20%22csv%22,%22url%22:%20%22test/incidents_notime.csv%22,%22name%22:%20%22start%20test%22}]}]}

Cheers,
Nick

**Reply by glughi** [reply]
Hi @nf-s,
I meant something similar to the example of

https://docs.terria.io/guide/deploying/controlling-with-url-parameters/#start-data

where `"init/nm.json", "init/test.json"` are specified into `initSources`.

Thanks,
G.

**Reply by nf-s** [reply]
Hi @glughi 

Oh i see!

Yes you are supposed to be able to do that - there is a bug

For example

```json
{
  "version": "8.0.0",
  "initSources": [
    "http://gist.githubusercontent.com/nf-s/67e44309245f3caca1cee5cdf2eee3eb/raw/75c50871fd972f8d6f35039c6e5278f3f7437a69/colorscale-minmax.json"
  ]
}
```

http://ci.terria.io/main/#start=%7B%0A%20%20%22version%22%3A%20%228.0.0%22%2C%0A%20%20%22initSources%22%3A%20%5B%0A%20%20%20%20%22http%3A%2F%2Fgist.githubusercontent.com%2Fnf-s%2F67e44309245f3caca1cee5cdf2eee3eb%2Fraw%2F75c50871fd972f8d6f35039c6e5278f3f7437a69%2Fcolorscale-minmax.json%22%0A%20%20%5D%0A%7D

Isn't working at the moment

I made a PR - I should be able to finish it next week https://github.com/TerriaJS/terriajs/pull/6402

**Reply by glughi** [reply]
Thanks a lot!
G.

**Reply by nicolasvila** [reply]
Hi, I would like to set up an intialization context at startup.
I'm currently the share button to get the current start context as "share date JSON".
What I would like to do is to load the JSON from the web storage at startup, without using /#start={....} How can I achieve this as the structure is different from Catalog/JSON ? 
Thanks for your help
